### PR TITLE
Bug 1373276 - Call POST /account/devices/notify in the send tab exten…

### PR DESCRIPTION
**Pull request by justindarc**
…sion

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
Profile.swift | 437 | Force casts should be avoided. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Providers/Profile.swift [hunk 1]
```diff
@@ -499,7 +499,18 @@ open class BrowserProfile: Profile {
         let commands = items.map { item in
             SyncCommand.displayURIFromShareItem(item, asClient: id)
         }
-        return self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
+        return self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> {
+            let syncStatus = self.syncManager.syncClients()
+            syncStatus >>> {
+                let deviceIDs = clients.filter({ $0.fxaDeviceId != nil }).map({ $0.fxaDeviceId! })
```
Comment (jhugman) [inline Providers/Profile.swift]:
```swift
clients.flatMap { $0.fxaDeviceId }
```

File: Providers/Profile.swift [hunk 2]
```diff
@@ -499,7 +499,18 @@ open class BrowserProfile: Profile {
...
+                guard let account = self.getAccount() else {
+                    return
+                }
+                
+                account.notify(deviceIDs: deviceIDs, collectionsChanged: ["clients"])
+            }
+            return syncStatus
+        }
```
Comment (jhugman) [inline Providers/Profile.swift]: This block needs tidying. 

Do we need to `notify` explicitly after `syncClients()` has returned? If so, comment.

Also, there's a `Deferred` inside a `Deferred`, is there anyway to tidy this?

File: Account/FxAClient10.swift [hunk 3]
```diff
@@ -52,6 +52,10 @@ public struct FxADevicesResponse {
     let devices: [FxADevice]
 }
 
+public struct FxANotifyResponse {
+    let success: Bool
+}
```
Comment (jhugman) [inline Account/FxAClient10.swift]: Is this necessary? Can't this just be a `Success`?

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
TestBookmarksRepairRequestor.swift | 82 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 93 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 205 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 270 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 283 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 331 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 
TestBookmarksRepairRequestor.swift | 337 | Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**